judge/admin/contest.py

```diff
@@ -148,7 +156,27 @@ def make_private(self, request, queryset):
 
     def get_urls(self):
         return [url(r'^rate/all/$', self.rate_all_view, name='judge_contest_rate_all'),
-                url(r'^(\d+)/rate/$', self.rate_view, name='judge_contest_rate')] + super(ContestAdmin, self).get_urls()
+                url(r'^(\d+)/rate/$', self.rate_view, name='judge_contest_rate'),
+                url(r'^(\d+)/judge/(\d+)/$', self.rejudge_view, name='judge_contest_rejudge'),
+                ] + super(ContestAdmin, self).get_urls()
+
+    def rejudge_view(self, request, contest_id, problem_id):
+        if not request.user.has_perm('judge.rejudge_submission'):
+            self.message_user(request, ugettext('You do not have the permission to rejudge submissions.'),
+                              level=messages.ERROR)
+            return
+        queryset = ContestSubmission.objects.filter(problem_id=problem_id)
```

Comment:
You might want to do a `select_related('submission')` or you are going to run a couple thousand queries. `harakiri` is not pretty when you managed to queue half of the submissions.